modules/core/include/opencv2/core/mat.inl.hpp

```diff
@@ -1096,6 +1096,12 @@ void Mat::push_back(const Mat_<_Tp>& m)
     push_back((const Mat&)m);
 }
 
+template<> inline
+void Mat::push_back(const MatExpr& expr)
+{
+    push_back((const Mat&)expr);
```

Comment:
nope. Mat_&lt;_Tp&gt; and Mat are binary identical, so the reference cast works just fine (and it's definitely faster)